subtree_compare.cpp

```diff
@@ -0,0 +1,150 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include <iostream>
+
+using namespace std;
+
+// basic node struct used in tree
+struct node {
+	node * left;
+	node * right;
+
+	int value;
+
+	node( int val ) : value(val), left(NULL), right(NULL) {}
+	node( int val, node * l, node * r ) : value(val), left(l), right(r) {}
+};
+
+// basic struct for holding tree
+struct Tree {
+	
+	node * root;
+	
+	Tree() : root(NULL) {}
+	void populate( int * num_arr, int n);
+};
+
+// recursively compares tree sub branches
+bool compare( node * t, node * s ) {
+	if ( t == NULL && s == NULL ) 
+		return true;
+	if( (t == NULL && s != NULL) || (t != NULL && s == NULL) )
```

Comment:
the s != null and t != null checks are redundant in this case, because if they weren't, then your previous condition would have already returned true. The check you really want is "if (t == null || s == null)"